### PR TITLE
B1' derived ef

### PR DESCRIPTION
**Pull request by jordidj**
## PR description
Added the derivative of B1 to the derived eigenfunctions to allow for computation of Delta' for tearing modes.

## New features
**Legolas**
- New derived ef: dB1

## Checklist
- [x] all tests pass (and for new features, new tests are added)
- [x] documentation has been updated (if applicable)

### CODE REVIEW

**Comment by jordidj** [comment]
To do:
[] regenerate derived ef tests
[] bump version number
[] add LaTeX form for dB1 to pylbo

**Comment by n-claes** [comment]
Poke me if this is ready and I'll give it a review. Failing tests can be simply fixed by bumping the expected number of derived eigenfunctions for the unit tests and regenerating the regression answers for the Couette and Taylor-Couette cases.
You can do this through
```bash
pytest test_taylor_couette.py --generate
```
and then it will ask you to confirm when the tests run. You can add the `--keep-files` flag as well so it doesn't automatically remove them after testing for visual inspection.

File: tests/regression_tests/regression.py [hunk 1]
```diff
@@ -309,8 +309,8 @@ def run_derived_eigenfunction_test(self, derived_eigenfunction, ds_test, ds_base
             ds_base,
             names_attr="derived_ef_names",
             figname_prefix="derived_",
-            nb_plots=(5, 4),
-            figsize=(18, 10),
+            nb_plots=(5, 5),
+            figsize=(23, 10),
```
Comment (n-claes) [inline tests/regression_tests/regression.py:313]: I assume this was increased because there are now 21 derived eigenfunctions instead of 20?

**Review by n-claes — APPROVED** [review]

**Comment by n-claes** [comment]
Looks good -- quick question, for testing purposes I assume that (at least) one of the regression tests you regenerated include a case in which `dB1` is non-zero? By this I mean a case where `dB1` properly behaves.

Merge when ready :)

File: tests/regression_tests/regression.py (hunk 1 above)
Comment (jordidj) [reply tests/regression_tests/regression.py:313]: Indeed. I don't know if setting it to (3, 7) to have exactly 21 plots would be better.

**Comment by jordidj** [comment]
`dB1` is nonzero and well-behaved in the MRI test case.

**Pull request merged**